packages/react-aria-components/stories/Tree.stories.tsx

```diff
@@ -1196,3 +1196,80 @@ export const TreeWithDragAndDropVirtualized = {
   render: TreeDragAndDropVirtualizedRender,
   name: 'Tree with drag and drop (virtualized)'
 };
+
+
+interface ITreeItem {
+  id: string,
+  name: string,
+  childItems?: Iterable<ITreeItem>
+}
+
+let totalItems = 0;
+let itemKeys = new Set<any>();
+/**
+ * Generates a tree data structure with 10 items per level and 6 levels deep.
+ * @returns Array of tree items with the specified structure.
+ */
+function generateTreeData(): Array<ITreeItem> {
+  /**
+   * Recursively generates tree items for a given level.
+   * @param level - Current depth level (1-6).
+   * @param parentId - Parent item ID for generating unique child IDs.
+   * @returns Array of tree items for this level.
+   */
+  function generateLevel(level: number, parentId: string = ''): Array<ITreeItem> {
+    const items: ITreeItem[] = [];
+    const itemsPerLevel = 7;
+
+    for (let i = 1; i < itemsPerLevel; i++) {
+      const itemId = parentId ? `${parentId}-${i}` : `level-${level}-item-${i}`;
+      const itemName = `Level ${level} Item ${i}`;
+
+      const item: ITreeItem = {
+        id: itemId,
+        name: itemName
+      };
+
+      // Add child items if not at the deepest level (level 6)
+      if (level < 6) {
+        item.childItems = generateLevel(level + 1, itemId);
+      }
+
+      totalItems++;
+      items.push(item);
+      itemKeys.add(itemId);
+    }
+
+    return items;
+  }
+
+  return generateLevel(1);
+}
+
+const treeData = generateTreeData();
+console.log(`Total items: ${totalItems}`);
```

Comment:
Best way to test this is to the root level `visitNode` in timing, I didn't include it so that the tests would all pass and we'd get a build, the delay/difference should be noticeable in the build, but just if you want numbers to put to it. 
```
  let startTime = Date.now();
  for (let node of collection) {
    visitNode(node);
  }
  let endTime = Date.now();
  console.log(`Time taken: ${endTime - startTime}ms`);
```